apps/checkout/src/hooks/useAlerts/useAlerts.tsx

```diff
@@ -0,0 +1,98 @@
+import { useErrorMessages } from "@/checkout/hooks/useErrorMessages";
+import {
+  MessageKey,
+  useFormattedMessages,
+} from "@/checkout/hooks/useFormattedMessages";
+import {
+  Alert,
+  AlertType,
+  AlertErrorData,
+  AlertSuccessData,
+  CheckoutScope,
+} from "./types";
+import { toast } from "react-toastify";
+import { camelCase } from "lodash-es";
+import { ApiErrors, useGetParsedApiErrors } from "@/checkout/hooks/useErrors";
+
+export interface ScopedAlertsProps {
+  showSuccess: () => void;
+  showErrors: (errors: AlertErrorData[]) => void;
+}
+
+function useAlerts(scope: CheckoutScope): ScopedAlertsProps;
+
+function useAlerts(): {
+  showSuccess: (scope: CheckoutScope) => void;
+  showErrors: (errors: AlertErrorData[], scope: CheckoutScope) => void;
+};
+
+function useAlerts(globalScope?: any): any {
```

Comment:
It returns an object every time so I think it could be typed better

Reply:
@Cloud11PL  This is intentional  - notice the function type definitions above. It's done because you can use this either like
```
const { showErrors } = useAlerts("userRegister");
...
showErrors(errors);
```
or
```
const { showErrors } = useAlerts();
...
showErrors(errors, "userRegister");
showErrors(otherErrors, "customerDetach");
```